### PR TITLE
Add a section about the UsageKind to the Service Catalog documentation

### PR DESCRIPTION
**Pull request by klaudiagrz**
Confirm these statements before you submit your pull request:

- [x] I have read and submitted the required [Contributor Licence Agreements](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#agreements-and-licenses).
- [x] This pull request follows the contributing guidelines, recommended Git workflow, and templates.
- [ ] I have tested my changes.
- [x] I have updated the relevant documentation.
---

**Description**

Changes proposed in this pull request:

- Add a section about the UsageKind 
- Add a diagram showing the UsageKind workflow

**Issue link**

### CODE REVIEW

File: docs/service-catalog/docs/012-details-provisioning-and-binding.md [hunk 1]
```diff
@@ -54,6 +54,12 @@ Kyma binding operation consists of two phases:
 
 > **NOTE:** The system allows you to create the ServiceBinding and ServiceBindingUsage resources at the same time.
 
+### Bind with other resources
+
+The UsageKind is a cluster-wide custom resource which allows you to bind a ServiceInstance to any resource. By default, Kyma provides two UsageKinds which enable binding either to a Deployment or Function. You can add more UsageKinds if you want to bind your ServiceInstance to the other custom resource. The UsageKind contains information on the way in which binding to this custom resource is conducted. The ServiceBindingUsage uses this information to inject Secrets to the Application.
```
Comment (tomekpapiernik) [inline docs/service-catalog/docs/012-details-provisioning-and-binding.md]: custom resource -> you're talking about a CR in the Kubernetes world, I'd capitalize it just as you would capitalize Custom Resource Definition.

Comment (tomekpapiernik) [inline docs/service-catalog/docs/012-details-provisioning-and-binding.md]: You can add more UsageKinds if you want to bind your ServiceInstance to the other custom resource -> (...) bind your ServiceInstance to other custom resources.

It's not like there's only one CR you can bind the service instance to, right?

Comment (klaudiagrz) [reply docs/service-catalog/docs/012-details-provisioning-and-binding.md]: Changed to: 
`to the other types of resources.`

Comment (klaudiagrz) [reply docs/service-catalog/docs/012-details-provisioning-and-binding.md]: As discussed, I wouldn't capitalize it.

**Pull request merged**